controllers/openstackbaremetalset_controller.go

```diff
@@ -1397,11 +1407,20 @@ func (r *OpenStackBaremetalSetReconciler) createNewHostnames(
 			return newHostnames, err
 		}
 
+		if instance.Status.BaremetalHosts == nil {
+			instance.Status.BaremetalHosts = map[string]ospdirectorv1beta1.HostStatus{}
+		}
```

Comment:
It looks like we're already doing this on lines 121-123.  Or maybe it is possible for the instance to arrive at the `createNewHostnames` func without hitting those lines above?

Reply:
yeah, I have still seen a assignment to entry in nil map error. need to debug why, or just add them here for safety